core/trino-main/src/main/java/io/trino/execution/RenameViewTask.java

```diff
@@ -86,7 +87,13 @@ public ListenableFuture<Void> execute(
             throw semanticException(CATALOG_NOT_FOUND, statement, "Target catalog '%s' does not exist", target.getCatalogName());
         }
         if (metadata.isView(session, target)) {
-            throw semanticException(TABLE_ALREADY_EXISTS, statement, "Target view '%s' already exists", target);
+            throw semanticException(GENERIC_USER_ERROR, statement, "Target view '%s' already exists", target);
+        }
+        if (metadata.getTableHandle(session, target).isPresent()) {
```

Comment:
The same comment for order of checks as above in `RenameTableTask`

Reply:
Done.